Add support for tuist graph to show a subset of targets

adellibovi opened the pull request:
Resolves https://github.com/tuist/tuist/issues/2433
Request for comments document (if applies):

### Short description 📝

This PR add the support of filtering targets for the command `tuist graph`, i.e `tuist graph FrameworkA`

### Checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase.
- [x] A developer other than the author has verified that the changes work as expected.
- [x] The changes have been tested following the [documented guidelines](https://tuist.io/docs/contribution/testing-strategy/).
- [x] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [x] In case the PR introduces changes that affect users, the documentation has been updated.

fortmarek commented on Tests/TuistGeneratorTests/GraphViz/GraphToGraphVizMapperTests.swift:
```diff
@@ -52,25 +38,27 @@ final class GraphToGraphVizMapperTests: XCTestCase {
         // When
-        let got = subject.map(graph: graph, skipTestTargets: false, skipExternalDependencies: true)
-        let expected = makeExpectedGraphViz(includeExternalDependencies: false)
+        let got = subject.map(graph: graph, skipTestTargets: false, skipExternalDependencies: true, targetsToFilter: ["Tuist iOS"])
```
would this test fail if the targets actually wouldn't be filtered? afaict it would not but maybe I am mistaken. Might be worth creating a target alongside `Tuist iOS`

adellibovi replied:
It should fail because of the “tuist watchos” which is at the same level as “tuist iOS”, right?

fortmarek replied:
ah, OK, I got sidetrack by `onlyiOS` below but now it makes sense, sorry 👍

fortmarek reviewed (approved):
I think this is a welcome addition as on projects with a lot of modules, it's really hard to find anything without filtering targets - thanks for doing this 👏

fortmarek commented:
seems swiftlint failed, try running `bundle exec rake style_correct`. 
Also I think it's worth making a fixture for the graph feature that would check that it successfully runs and `graph.png` is created. I don't think there is more we could test in a fixture (thoughts @natanrolnik?) - anyways, if you don't mind, it'd be nice to include it in this PR. I think it should not take too much time to implement. Otherwise I can try to take a stab at it in a different PR.

fortmarek reviewed (commented):
I have now tried running it `tuist graph -d -t` and Preview tells me the file could not be opened because "It may be damaged or use a file format that Preview doesn’t recognise." - I can not tell if it is because of your changes or if the change is there for a little longer, I'll investigate it further tomorrow. But definitely would be good to spin up an acceptance test for this.

adellibovi commented:
Added already a couple of acceptance tests 😄 
I had to make a change at `tuist graph` command too, we were using `--path` to define the output of the file, while for every other command `--path` define where the manifest is, I changed accordingly and added an `--output-path`, updated both changelog and documentation.
Let me know what you find out with `-t -d`, I tested both on some fixture and on my project and went fine

kwridan reviewed (approved):
Thanks for this @adellibovi - just tested it out, it looks great :)

fortmarek commented:
It is probably problem on my side - might be related that I am now temporarily on Catalina, I'll try to update the system this weekend. I think it is good to merge 🙂
